pkg/daemon/drain.go

```diff
@@ -115,13 +115,16 @@ func (dn *Daemon) performDrain() error {
 }
 
 // isDrainRequired determines whether node drain is required or not to apply config changes.
-func isDrainRequired(actions, diffFileSet []string, oldIgnConfig, newIgnConfig ign3types.Config) (bool, error) {
+func isDrainRequired(actions, diffFileSet []string, oldIgnConfig, newIgnConfig ign3types.Config, overrideImageRegistryDrain bool) (bool, error) {
 	if ctrlcommon.InSlice(postConfigChangeActionReboot, actions) {
 		// Node is going to reboot, we definitely want to perform drain
 		return true, nil
 	} else if ctrlcommon.InSlice(postConfigChangeActionReloadCrio, actions) || ctrlcommon.InSlice(postConfigChangeActionRestartCrio, actions) {
 		// Drain may or may not be necessary in case of container registry config changes.
 		if ctrlcommon.InSlice(constants.ContainerRegistryConfPath, diffFileSet) {
+			if overrideImageRegistryDrain {
```

Comment:
Since this is a risky operation and it is possible that admin may forget about the configmap later on. we would want to add a warning log, raise an event and perhaps alert as well to help better troubleshoot bug later.

Reply:
If warning functionality is added, then the ability to disable the warning in the config map should exist as well.   We want to reduce unnecessary logging when not desired.

Reply:
Added a log here for troubleshooting. I opted not to add any alerting since this should only be used as a support exception. The MCO itself also would not fail regardless, just potential pods not spinning up, and those should have the actual error re: not pullable images.

The MCO isn't actually changing any error methods here. The MCO is only changing when an error is surfaced, and the core failure model stays the same.

Reply:
Eventually with admin defined reboots existing, I think we can retroactively add a alert in a future version for users to move to the API method instead of having this, if they still have this in 4.18